src/components/Trade/TradeInput.tsx

```diff
@@ -200,29 +210,53 @@ export const TradeInput = () => {
     [updateAction, updateIsSendMax, updateAmount, handleInputAmountChange],
   )
 
-  const handleSendMax: TradeAssetInputProps['onPercentOptionClick'] = useCallback(() => {
+  const handleSendMax: TradeAssetInputProps['onPercentOptionClick'] = useCallback(async () => {
     if (!(sellAsset && activeQuote)) return
-    const maxSendAmount = getSendMaxAmount(
-      sellAsset,
-      sellFeeAsset,
-      activeQuote,
-      sellAssetBalanceCrypto,
-    )
-    updateSellAmountCryptoPrecision(maxSendAmount)
+
+    // Network fee is a function of the sell amount, so we need a quote for an arbitrarily high sell amount to ensure
+    // we get a workable max amount
+    const availableSwapperTypesWithQuoteMetadata =
+      tradeQuoteArgs && feeAsset
+        ? (
+            await dispatch(
+              getAvailableSwappers.initiate({
+                ...tradeQuoteArgs,
+                sellAmountBeforeFeesCryptoBaseUnit: '10000000', // arbitrarily high sell amount for max send quote
```

Comment:
How reliable is this? Depending on the base unit, that may not be so high of an amount. Regardless, slippage and outbound fees are bound to vary depending on the amounts.
Now that we get a proper rate for the amount before fees and slippage, can we simply pass the input sell amount in base unit? see the PR I've sneaked under this https://github.com/shapeshift/web/pull/4221

Reply:
We can't use the input sell amount, because if it's 0 (i.e. on initial load and the user hasn't entered any amounts yet), we get smaller network fees, resulting in a computed max sell amount that is too high (and thus the scenario where it needs to be pressed twice to get an accurate amount).

My comment is actually misleading - it doesn't really need to be high, it really just needs to not be `0`, as the fees and gas for some swap endpoints seem to be less when it's `0`. Anything above `0` seems to return the same amount (at least for the swappers that we use, though this is not guaranteed).

That said, with the change to the THORChain swapper endpoint, the amount is no longer arbitrary, as it needs to be high enough to actually get a quote back (as now we get nothing if we don't meet the minimum).

So, perhaps more thought it needed here.

Reply:
Actually, we might be able to set it to the theoretical max value based on the sell asset's precision, though we'll get rugged if a swappers network or trading fees are ever a function of the sell amount.

Reply:
Yeah looks like a catch-22 of sorts, we need an amount to get an initial *valid* quote, but we don't necessarily have an amount to start with, and need a quote to get an amount depending on the rate. Confirmed `getSendMaxAmount()` is doing its job so I'm happy with this.

Gave it some good testing with my testing wallet and it seems to do the trick although I've really tested on small amounts.

Gave it a second pass on `vitalik.eth`'s wallet as a watch-only Frame wallet, with larger amounts:

<img width="976" alt="image" src="https://user-images.githubusercontent.com/17035424/231121669-dffa001d-2a80-4ddd-a5dd-a45e4aa0bf22.png">
<img width="630" alt="image" src="https://user-images.githubusercontent.com/17035424/231121977-b13ea12f-e590-4e02-9a3b-983234490ca9.png">

<img width="1633" alt="image" src="https://user-images.githubusercontent.com/17035424/231122883-b06953e4-2ca0-4261-a06f-fe4931c6f04f.png">
<img width="601" alt="image" src="https://user-images.githubusercontent.com/17035424/231122643-6238d2bd-afd9-41d4-8abc-bd5b92b7e375.png">

That LGTM!